src/com/facebook/buck/cxx/DefaultCxxPlatforms.java

```diff
@@ -42,15 +43,44 @@ private DefaultCxxPlatforms() { }
 
   public static final Flavor FLAVOR = ImmutableFlavor.of("default");
 
-  private static final Path DEFAULT_C_FRONTEND = Paths.get("/usr/bin/gcc");
-  private static final Path DEFAULT_CXX_FRONTEND = Paths.get("/usr/bin/g++");
-  private static final Path DEFAULT_AR = Paths.get("/usr/bin/ar");
-  private static final Path DEFAULT_STRIP = Paths.get("/usr/bin/strip");
-  private static final Path DEFAULT_RANLIB = Paths.get("/usr/bin/ranlib");
-  private static final Path DEFAULT_NM = Paths.get("/usr/bin/nm");
+  private static final Path DEFAULT_C_FRONTEND;
+  private static final Path DEFAULT_CXX_FRONTEND;
+  private static final Path DEFAULT_AR;
+  private static final Path DEFAULT_STRIP;
+  private static final Path DEFAULT_RANLIB;
+  private static final Path DEFAULT_NM;
 
-  private static final Path DEFAULT_OSX_C_FRONTEND = Paths.get("/usr/bin/clang");
-  private static final Path DEFAULT_OSX_CXX_FRONTEND = Paths.get("/usr/bin/clang++");
+  private static final Path DEFAULT_OSX_C_FRONTEND;
+  private static final Path DEFAULT_OSX_CXX_FRONTEND;
+
```

Comment:
Rather than a static initialiser, let's look up these paths inside build. Specifically, these are platform-specific, and not all of them need to exist on all platforms, so let's not try to look then up if we're not going to use them.